fun test() {
    <!INAPPLICABLE_CANDIDATE!>Double<!>()
    <!INAPPLICABLE_CANDIDATE!>Float<!>()
    <!INAPPLICABLE_CANDIDATE!>Long<!>()
    <!INAPPLICABLE_CANDIDATE!>Int<!>()
    <!INAPPLICABLE_CANDIDATE!>Short<!>()
    <!INAPPLICABLE_CANDIDATE!>Byte<!>()
    <!INAPPLICABLE_CANDIDATE!>Char<!>()
    <!INAPPLICABLE_CANDIDATE!>Boolean<!>()
}