class A {
    fun bar() {
        <!AMBIGUITY!>div<!> {
            <!AMBIGUITY!>div<!> {
                <!AMBIGUITY!>div<!> {
                    <!AMBIGUITY!>div<!> {
                        <!AMBIGUITY!>div<!> {
                            <!AMBIGUITY!>div<!> {
                                <!AMBIGUITY!>div<!> {
                                    <!AMBIGUITY!>div<!> {
                                        <!AMBIGUITY!>div<!> {
                                            <!AMBIGUITY!>div<!> {
                                                <!AMBIGUITY!>div<!> {
                                                    <!AMBIGUITY!>div<!> {
                                                        <!AMBIGUITY!>div<!> {
                                                            <!AMBIGUITY!>div<!> {
                                                                <!AMBIGUITY!>div<!> {
                                                                    <!AMBIGUITY!>div<!> {
                                                                        <!AMBIGUITY!>div<!> {
                                                                            <!AMBIGUITY!>div<!> {
                                                                                <!AMBIGUITY!>div<!> {
                                                                                    <!AMBIGUITY!>div<!> {
                                                                                        <!AMBIGUITY!>div<!> {
                                                                                            <!AMBIGUITY!>div<!> {
                                                                                                <!AMBIGUITY!>div<!> {
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                    <!UNRESOLVED_REFERENCE!>+<!>(<!UNRESOLVED_REFERENCE!>foo<!> ?: "")
                                                                                                }
                                                                                            }
                                                                                        }
                                                                                    }
                                                                                }
                                                                            }
                                                                        }
                                                                    }
                                                                }
                                                            }
                                                        }

                                                    }
                                                }
                                            }
                                        }
                                    }
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
