// !DIAGNOSTICS: -REDECLARATION

package<!SYNTAX!><!>

fun () {

}

val<!SYNTAX!><!> : Int = 1


class<!SYNTAX!><!> {

}

interface<!SYNTAX!><!> {

}

object<!SYNTAX!><!> {

}

enum class<!SYNTAX!><!> {}

annotation class<!SYNTAX!><!> {}
